contracts/utils/VaultQueue.sol

```diff
@@ -0,0 +1,228 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.4;
+
+import {
+    Initializable
+} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import {
+    OwnableUpgradeable
+} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {
+    ReentrancyGuardUpgradeable
+} from "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {
+    SafeERC20
+} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {ISAVAX} from "../interfaces/ISAVAX.sol";
+import {IRibbonVault, IDepositContract} from "../interfaces/IRibbon.sol";
+import {ICRV} from "../interfaces/ICRV.sol";
+import {Vault} from "../libraries/Vault.sol";
+import {DSMath} from "../vendor/DSMath.sol";
+
+contract VaultQueue is
+    Initializable,
+    OwnableUpgradeable,
+    ReentrancyGuardUpgradeable
+{
+    enum TransferAction {INTERVAULT, WITHDRAW}
+    struct Transfer {
+        address creditor;
+        address srcVault;
+        address dstVault;
+        address depositContract;
+        TransferAction action;
+        uint256 amount;
+        uint256 timestamp;
+    }
+
+    using SafeERC20 for IERC20;
+
+    mapping(address => Transfer[]) public qTransfer;
+    mapping(address => uint256) public totalAmount;
+
+    address public wethVault;
+    address public stethVault;
+    address public stethEthCrvPool;
+
+    event Disburse(address vault, Transfer txn, uint256 portion);
+
+    function initialize(
+        address _wethVault,
+        address _stethVault,
+        address _stethEthCrvPool
+    ) external initializer {
+        __ReentrancyGuard_init();
+        __Ownable_init();
+
+        transferOwnership(msg.sender);
+
+        wethVault = _wethVault; // Can be the native vault (weth or wavax)
+        stethVault = _stethVault; // Must ONLY be steth vault.  (savax vault returns erc20, set as keeper)
+        stethEthCrvPool = _stethEthCrvPool;
```

Comment:
We're not using the `stethEthCrvPool` as part of the withdrawal so we can remove this entirely